core/txpool/list.go

```diff
@@ -571,6 +598,7 @@ func (l *pricedList) Discard(slots int, force bool) (types.Transactions, bool) {
 			tx := heap.Pop(&l.urgent).(*types.Transaction)
 			if l.all.GetRemote(tx.Hash()) == nil { // Removed or migrated
 				atomic.AddInt64(&l.stales, -1)
+				slots -= numSlots(tx)
```

Comment:
Are you sure the slots for stale transcations aren't already removed from the count? I am not, off the top of my head

Reply:
When it was removed from `l.all`: 
```
// Remove removes a transaction from the lookup.
func (t *lookup) Remove(hash common.Hash) {
	t.lock.Lock()
	defer t.lock.Unlock()

	tx, ok := t.locals[hash]
	if !ok {
		tx, ok = t.remotes[hash]
	}
	if !ok {
		log.Error("No transaction found to be deleted", "hash", hash)
		return
	}
	t.slots -= numSlots(tx)
```
Then we stopped counting it's slots. So it's not fair to, at this point, credit the number of slots to the new tx. We've already freed those slots.

Reply:
So afaict this change should be reverted